### PR TITLE
Address select

### PR DESCRIPTION
**Pull request by kielbarry**
#### What does it do?
This is a dropdown menu for wallet and contract addresses (each displayed slightly differently). it is pre-populated with the address that has the highest ether balance. Column css is applied at a div layer higher than this component. Stories include both providing, wallets and contracts, one or the other, or an empty case.

#### Any helpful background information?
This PR is in reference to Issue #79.

#### New dependencies? What are they used for?
shortid is used to generate unique keys for items in a list.

#### Relevant screenshots?
screen shot 2019-01-14 at 3 51 40 pm

### CODE REVIEW

File: src/components/Widgets/Form/AddressSelect.jsx [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+              </StyledOption>
+            );
+          })}
+        </StyledSelect>
+        <StyledIdenticon
+          title
+          size={identiconSize}
+          address={selectedWallet}
+        />
+        </StyledDiv>
```
Comment (marcgarreau) [inline src/components/Widgets/Form/AddressSelect.jsx]: Style nit: indentation is off here. If you use `prettier` in your editor, this shouldn't be an issue. I've just made (and self-assigned) a new task to add prettier as a commit hook to auto-fix this sort of thing in the future.

File: src/stories/1.index.stories.jsx [hunk 2]
```diff
@@ -21,10 +21,99 @@ import {
...
 storiesOf('Welcome', module).add('to Ethereum Components', () => (
   <Welcome showApp={linkTo('Button')} />
 ))
 
+storiesOf('Widgets/Form/AddressSelect', module)
+  .add('default', () => (
+    <AddressSelect
+      wallets={dummyWallets}
+      walletContracts={dummyContracts}
+      onChange={() => alert('onChange registered')}
```
Comment (marcgarreau) [inline src/stories/1.index.stories.jsx]: The alerts get intrusive in the storybook. Instead, just pass an empty function, e.g. `onChange={() => {}}`

File: src/stories/1.index.stories.jsx [hunk 3]
```diff
@@ -21,10 +21,99 @@ import {
...
+    balance: '0.0',
+    name: 'Account 7',
+    addressType: 'wallet',
+  },
+}
+// contract dummy data
+const dummyContracts =
+  {
+    '0xD26B16d9Cf2054fd0B266a03A11C4CC198Eed025': {
+    // contractAddress: '0xD26B16d9Cf2054fd0B266a03A11C4CC198Eed025',
```
Comment (marcgarreau) [inline src/stories/1.index.stories.jsx]: nit: remove comment

File: src/stories/1.index.stories.jsx [hunk 4]
```diff
@@ -21,10 +21,99 @@ import {
...
+    name: 'Account 3',
+    addressType: 'wallet',
+  },
+   '0xghi5F863Ddb456601783A57A1C3E9F8f3ebc6c3B': {
+    balance: 0.0,
+    name: 'Account 4',
+    addressType: 'wallet',
+  },
+  '0xzzz5F863Ddb456601783A57A1C3E9F8f3ebc6c3B': {
+    balance: 0.01,
```
Comment (marcgarreau) [inline src/stories/1.index.stories.jsx]: nit: some values as numbers, some as strings. Please have all be strings.

File: src/components/Widgets/Form/AddressSelect.jsx [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+  display: block;
+  white-space: pre;
+  min-height: 1.2em;
+  padding: 0px 2px 1px;
+  color: #02a8f3;
+  line-height: 18px;
+  font-size: 1em;
+`
+
+const StyledSelect = styled.select`
```
Comment (marcgarreau) [inline src/components/Widgets/Form/AddressSelect.jsx]: Did you try out the `react-select` component? Was there a reason you didn't use it?

File: src/components/Widgets/Form/AddressSelect.jsx [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+                &nbsp; - &nbsp;
+                { w.balance }
+                &nbsp;
+                ETHER
+              </StyledOption>
+            );
+          })}
+        </StyledSelect>
+        <StyledIdenticon
+          title
```
Comment (marcgarreau) [inline src/components/Widgets/Form/AddressSelect.jsx]: nit: title is unused

File: src/components/Widgets/Form/AddressSelect.jsx [hunk 7]
```diff
@@ -0,0 +1,134 @@
...
+import { combineWallets } from '../../../util/helpers';
+
+export default class AddressSelect extends Component {
+  static displayName = 'AddressSelect';
+
+  static propTypes = {
+    disabled: PropTypes.bool,
+    wallets: PropTypes.object,
+    walletContracts: PropTypes.object,
+    identiconSize: PropTypes.string,
```
Comment (marcgarreau) [inline src/components/Widgets/Form/AddressSelect.jsx]: I think we can remove the option to set `Identicon` size, and hard code it as `tiny`. I don't believe this component is used elsewhere or needs this type of configurability. Correct me if I'm wrong.

File: src/stories/1.index.stories.jsx (hunk 4 above)
Comment (ryanio) [reply src/stories/1.index.stories.jsx]: yes, and you can try small values like ".000001" ether or even in hexadecimal format  to verify it renders correctly. when passed into BN like `new BN(wallet.balance).toString()` usually normalizes the output pretty well

File: src/util/helpers.js [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+    .filter(wallet => parseFloat(wallet.balance) !== 0)
+    .sort((a, b) => b.balance - a.balance);
+}
+
+// export function combineWallets(addresses, walletContracts) {
+export function combineWallets(addr, wc) {
+  /* scrub for undefined parameters */
+  let addresses = addr;
+  let walletContracts = wc;
+  if (!addr) addresses = {}
```
Comment (ryanio) [inline src/util/helpers.js]: nit: i don't love one-liner ifs, i always prefer to have curly braces around, @marcgarreau what do you think

File: src/components/Widgets/Form/AddressSelect.jsx (hunk 7 above)
Comment (ryanio) [reply src/components/Widgets/Form/AddressSelect.jsx]: agreed, but i do like your initiative kiel of adding configurable options :) can be useful, but agree with marc let's do it when we find the need

File: src/components/Widgets/Form/AddressSelect.jsx (hunk 1 above)
Comment (marcgarreau) [reply src/components/Widgets/Form/AddressSelect.jsx]: fyi - just merged a PR containing the prettier hook; please rebase on master. you may need to blow away your `yarn.lock` and do a fresh `yarn`.

File: src/components/Widgets/Form/AddressSelect.jsx [hunk 9]
```diff
@@ -0,0 +1,134 @@
...
+    return (
+      <StyledDiv>
+        <StyledSelect
+          onChange={this.selectAddress}
+          disabled={disabled}
+          value={selectedWallet}
+        >
+          {combinedWallets.map(w => {
+            return (
+              <StyledOption key={shortid.generate()} value={w.address}>
```
[the remark this answers is not in the document]
Comment (kielbarry) [reply src/components/Widgets/Form/AddressSelect.jsx]: Done.

File: src/components/Widgets/Form/AddressSelect.jsx (hunk 1 above)
Comment (kielbarry) [reply src/components/Widgets/Form/AddressSelect.jsx]: ^^ I will pull.

File: src/util/helpers.js (hunk 8 above)
Comment (kielbarry) [reply src/util/helpers.js]: I like one-liners and usually reserve them for 'gatekeeping' at the top of functions - just a style I prefer and have picked up along the way. I've removed them from this PR.

File: src/components/Widgets/Form/AddressSelect.jsx [hunk 10]
```diff
@@ -0,0 +1,134 @@
...
+                {
+                  w.addressType === 'wallet'
+                    ? `🔑 ${w.name}`
+                    : w['contract-name']
+                }
+                &nbsp; - &nbsp;
+                { w.balance }
+                &nbsp;
+                ETHER
```
[the remark this answers is not in the document]
Comment (kielbarry) [reply src/components/Widgets/Form/AddressSelect.jsx]: Done.

File: src/stories/1.index.stories.jsx (hunk 3 above)
Comment (kielbarry) [reply src/stories/1.index.stories.jsx]: Done.

File: src/stories/1.index.stories.jsx (hunk 4 above)
Comment (kielbarry) [reply src/stories/1.index.stories.jsx]: Done.

File: src/components/Widgets/Form/AddressSelect.jsx (hunk 5 above)
Comment (kielbarry) [reply src/components/Widgets/Form/AddressSelect.jsx]: Just wanted to make sure to get in a PR on Monday, but I've just pushed a version using react-select.

File: src/components/Widgets/Form/AddressSelect.jsx [hunk 11]
```diff
@@ -0,0 +1,148 @@
...
+  }
+
+  render() {
+    const { disabled } = this.props;
+    const { combinedWallets, selectedWallet } = this.state
+
+    let options = combinedWallets.map(w => {
+      return {
+        value: w.address,
+        label :w.addressType === 'wallet'
```
Comment (marcgarreau) [inline src/components/Widgets/Form/AddressSelect.jsx]: You've still got formatting and linting issues still hanging around. Please, please pause to integrate `prettier` into your editor (https://prettier.io/docs/en/editors.html) and use the linter in your editor or run `yarn lint:watch` in your terminal to spot lints in files you've worked in. Also, looks like you didn't actually rebase or merge master into your work, so recreating the yarn.lock file didn't have the desired effect of resolving conflicts.

File: src/components/Widgets/Form/AddressSelect.jsx [hunk 12]
```diff
@@ -0,0 +1,148 @@
...
+  render() {
+    const { disabled } = this.props;
+    const { combinedWallets, selectedWallet } = this.state
+
+    let options = combinedWallets.map(w => {
+      return {
+        value: w.address,
+        label :w.addressType === 'wallet' 
+          ? `🔑 ${w.name}` + ` - ${w.balance} ETHER`
+          : w['contract-name'] + ` - ${w.balance} ETHER`,
```
Comment (marcgarreau) [inline src/components/Widgets/Form/AddressSelect.jsx]: You're still mixing template literals with string concatenation here. The whole line can just be a template literal.

**Pull request closed without merging**